server/sonar-qa-util/src/main/java/org/sonarqube/qa/util/WebhookTester.java

```diff
@@ -0,0 +1,133 @@
+import org.sonarqube.ws.client.webhooks.DeliveriesRequest;
+import org.sonarqube.ws.client.webhooks.DeliveryRequest;
+import org.sonarqube.ws.client.webhooks.SearchRequest;
+import org.sonarqube.ws.client.webhooks.WebhooksService;
+
+import static java.util.Arrays.stream;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class WebhookTester {
+  private static final AtomicInteger ID_GENERATOR = new AtomicInteger();
+
+  private final TesterSession session;
+
+  WebhookTester(TesterSession session) {
+    this.session = session;
+  }
+
+  public WebhooksService service() {
+    return session.wsClient().webhooks();
+  }
+
+  public Webhook generate(Consumer<CreateRequest>... populators) {
+    return generate(null, null, populators);
+  }
+
+  public Webhook generate(Organization organization, Consumer<CreateRequest>... populators) {
+    return generate(organization, null, populators);
+  }
+
+  public Webhook generate(Project project, Consumer<CreateRequest>... populators) {
+    return generate(null, project, populators);
+  }
+
+  @SafeVarargs
+  public final Webhook generate(
+    @Nullable Organization organization,
+    @Nullable Project project,
+    Consumer<CreateRequest>... populators
+  ) {
+    int id = ID_GENERATOR.getAndIncrement();
+    CreateRequest request = new CreateRequest()
+      .setName("Webhook " + id)
+      .setUrl("https://webhook-" + id)
+      .setProject(project != null ? project.getKey(): null)
+      .setOrganization(organization != null ? organization.getKey() : null);
+    stream(populators).forEach(p -> p.accept(request));
+    return service().create(request).getWebhook();
+  }
+
+  public void deleteAll() {
+    deleteAll(null, null);
+  }
+
+  public void deleteAll(Organization organization) {
+    deleteAll(organization, null);
+  }
+
+  public void deleteAll(Project project) {
+    deleteAll(null, project);
+  }
+
+  public final void deleteAll(@Nullable Organization organization, @Nullable Project project) {
+    SearchRequest request = new SearchRequest()
+      .setOrganization(organization != null ? organization.getKey() : null)
+      .setProject(project != null ? project.getKey() : null);
+    service().search(request).getWebhooksList().forEach(p -> service().delete(new DeleteRequest().setKey(p.getKey())));
+  }
+
+  public List<Delivery> getPersistedDeliveries(Project project) {
+    DeliveriesRequest deliveriesReq = new DeliveriesRequest().setComponentKey(project.getKey());
+    return service().deliveries(deliveriesReq).getDeliveriesList();
+  }
+
+  public Delivery getPersistedDeliveryByName(Project project, String webhookName) {
+    List<Delivery> deliveries = getPersistedDeliveries(project);
+    Optional<Delivery> delivery = deliveries.stream().filter(d -> d.getName().equals(webhookName)).findFirst();
+    assertThat(delivery.isPresent()).isTrue();
+    return delivery.get();
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Call "delivery.isPresent()" before accessing the value. ([squid:S3655](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2FMMF-647%2FSONAR-10347_deliveries&issues=AWGPD6PDFMFpNJFufv6B&open=AWGPD6PDFMFpNJFufv6B)

[ISSUE_KEY]: AWGPD6PDFMFpNJFufv6B